[Win] Remove the code rounding off glyph advances and offsets

fujii opened the pull request:
#### 7ec79836337c20eaff56aa3600437621e2e013f1
<pre>
[Win] Remove the code rounding off glyph advances and offsets
<a href="https://bugs.webkit.org/show_bug.cgi?id=228536">https://bugs.webkit.org/show_bug.cgi?id=228536</a>

Reviewed by Don Olmstead.

Since <a href="https://commits.webkit.org/18359@main">https://commits.webkit.org/18359@main</a> added the code integrating
with Uniscribe for Windows port, it was rounding off glyph advances
and offsets.

However, this caused layout issues in some web pages especially with
high DPI display.

isSystemFont() method of FontPlatformData class is no longer used for
Windows port. Removed it.

* LayoutTests/platform/wincairo/TestExpectations:
* Source/WebCore/platform/graphics/FontPlatformData.h:
* Source/WebCore/platform/graphics/win/ComplexTextControllerUniscribe.cpp:
* Source/WebCore/platform/graphics/win/FontPlatformDataWin.cpp:
* Source/WebCore/platform/graphics/win/cairo/FontPlatformDataWinCairo.cpp:

Canonical link: <a href="https://commits.webkit.org/280214@main">https://commits.webkit.org/280214@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/31104268a65802f4285be7cb64569c1b70f9cf9f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/55961 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/35285 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/8429 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/58960 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/6393 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/58087 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/42906 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/6589 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/45065 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/4418 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/57990 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/33179 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/48253 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/26203 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/29962 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/5581 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/4536 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/51938 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/5852 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/60547 "Built successfully") | 
| | [⏳ 🛠 vision ](https://ews-build.webkit.org/#/builders/visionOS-1-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/5972 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/52492 "Passed tests") | 
| | [⏳ 🛠 vision-sim ](https://ews-build.webkit.org/#/builders/visionOS-1-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/48322 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/52018 "Passed tests") | 
| | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-1-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8287 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/31114 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/32198 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/33280 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/31946 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

donny-dont commented on Source/WebCore/platform/graphics/FontPlatformData.h:
```diff
@@ -301,7 +301,7 @@ class FontPlatformData {
 
-#if PLATFORM(WIN) || PLATFORM(COCOA)
+#if PLATFORM(COCOA)
     bool isSystemFont() const { return m_isSystemFont; }
```
Should this just be moved into the `USE(CORE_TEXT)` block above?

fujii replied:
`isSystemFont()` is used from `PLATFORM(COCOA)` code block too.
https://github.com/WebKit/WebKit/blob/535d9cfa7e4a6c1cce863562cf29272717bdb1b1/Source/WebCore/rendering/RenderBlock.cpp#L3537

donny-dont reviewed (approved):
Windows bits are fine. Just not sure if the definition should be moved around for 🍎 platforms.

fujii committed:
[Win] Remove the code rounding off glyph advances and offsets
https://bugs.webkit.org/show_bug.cgi?id=228536

Reviewed by Don Olmstead.

Since https://commits.webkit.org/18359@main added the code integrating
with Uniscribe for Windows port, it was rounding off glyph advances
and offsets.

However, this caused layout issues in some web pages especially with
high DPI display.

isSystemFont() method of FontPlatformData class is no longer used for
Windows port. Removed it.

* LayoutTests/platform/wincairo/TestExpectations:
* Source/WebCore/platform/graphics/FontPlatformData.h:
* Source/WebCore/platform/graphics/win/ComplexTextControllerUniscribe.cpp:
* Source/WebCore/platform/graphics/win/FontPlatformDataWin.cpp:
* Source/WebCore/platform/graphics/win/cairo/FontPlatformDataWinCairo.cpp:

Canonical link: https://commits.webkit.org/280214@main
